test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/ValidateTests.cs

```diff
@@ -139,6 +139,241 @@ public async void GivenAValidateByIdRequest_WhenTheResourceIdIsMissing_ThenADeta
                 "Patient.id");
         }
 
+        [Fact]
+        public async void GivenAValidateRequest_WhenAValidResourceIsPassedByParameter_ThenAnOkMessageIsReturned()
+        {
+            var payload = "{\"resourceType\": \"Parameters\", \"parameter\": [{\"name\": \"resource\", \"resource\": {\"resourceType\": \"Patient\", \"id\": \"123\"}}]}";
+
+            OperationOutcome outcome = await _client.ValidateAsync("Patient/$validate", payload);
+
+            Assert.Single(outcome.Issue);
+            CheckOperationOutcomeIssue(
+                outcome.Issue[0],
+                OperationOutcome.IssueSeverity.Information,
+                OperationOutcome.IssueType.Informational,
+                Success);
+        }
+
+        [Fact]
+        public async void GivenAValidateRequest_WhenAnInvlaidResourceIsPassedByParameter_ThenADetailedErrorIsReturned()
+        {
+            var payload = "{\"resourceType\": \"Parameters\", \"parameter\": [{\"name\": \"resource\", \"resource\": {\"resourceType\":\"Patient\",\"name\":{\"family\":\"test\",\"given\":\"one\"}}}]}";
+
+            OperationOutcome outcome = await _client.ValidateAsync("Observation/$validate", payload);
+
+            Assert.Single(outcome.Issue);
+            CheckOperationOutcomeIssue(
+                outcome.Issue[0],
+                OperationOutcome.IssueSeverity.Error,
+                OperationOutcome.IssueType.Invalid,
+                "Resource type in the URL must match resourceType in the resource.",
+                "TypeName");
+        }
+
+        [Theory]
+        [InlineData("CREATE", "CREATE is not a supported validation mode.", OperationOutcome.IssueType.NotSupported)]
+        [InlineData("UPDATE", "Resources can not be validated for update or delete at the default endpoint. An id must be provided in the URL.", OperationOutcome.IssueType.Invalid)]
+        [InlineData("DELETE", "Resources can not be validated for update or delete at the default endpoint. An id must be provided in the URL.", OperationOutcome.IssueType.Invalid)]
+        [InlineData("invalid", "invalid is not a valid validation mode.", OperationOutcome.IssueType.Invalid)]
+        public async void GivenAValidateRequest_WhenAModeIsPassed_ThenAnExceptionIsReturned(string mode, string message, OperationOutcome.IssueType issueType)
+        {
+            var payload = "{\"resourceType\": \"Patient\", \"id\": \"123\"}";
+
+            try
+            {
+                await _client.ValidateAsync("Patient/$validate?mode=" + mode, payload);
```

Comment:
We generally use `Assert.ThrowsAsync<FhirException>()` when trying to validate that an exception is thrown. This also avoids having the `Assert.True(false)` below (in order to fail the test if the exception is not thrown). You can use something like `FhirException ex = await Assert.ThrowsAsync<FhirException>()` and then validate on `ex`.

Reply:
Changed